lib/base_service.ts

```diff
@@ -52,18 +53,19 @@ export interface Credentials {
   password: string;
   api_key: string;
   url: string;
+  access_token: string;
```

Comment:
I know you were following the pattern here, but this interface doesn't quite seem right. It seems odd that all of these values are required instead of optional.

Reply:
I agree. With different methods of authentication allowed, it doesn't seem to fit the general case to make these all required. That being said, is has not caused any type-issues despite never including all of these.

@germanattanasio or @anweshan can you comment on this?

Reply:
@glennrfisher you're right that it looks odd 🥇 . I think it still works because although those all/some of those keys might have null/undefined values, as long as they're still keys on the object, Typescript doesn't complain ? That's my guess from looking at the methods that return `Credentials`.

Reply:
It hasn't caused any type issues because we are never calling the Credentials interface before it has these values. We instead call things like getCredentials that return an Credentials interface with all these options, which are blank if nothing was there. And then we have methods that call upon getCredentials.username to see if there was a username provided, etc. If the parameters are optional, then I think that when getCredentials() creates a new Credentials interface, it then won't have all the parameters like username, which could lead to exceptions calling for getCredentials.username etc. when not provided.

Reply:
What do you mean by blank? I had not looked into this much so I am testing out the current behavior. The following code:
```js
const conversation = new ConversationV1({
  url: myUrl,
  version: '2017-05-26',
  access_token: token
});

const credentials = conversation.getCredentials();
console.log(JSON.stringify(credentials, null, 2));
```
produces the output:
```
{
  "url": "<the url>",
  "access_token": "<the token>"
}
```
`credentials.username` returns `undefined`, the same as `credentials.foobar` or any other key that doesn't exist on the object. I am not understanding what the type-safety around requiring all of those keys is granting us.

Reply:
Okay, in that case you are right. I wasn't sure if in TS it will cause an error calling things like `credentials.fodder` in our code, but it does not. In that case it really doesn't matter if the keys are required or not required, since we only call the interface through `getCredentials()` rather than the interface itself...so you can change it to be optional for more clarity, although it won't affect the code itself.

Reply:
Okay, that makes sense. Thanks. This PR is merged anyways but I might make that change in the next